test/api/v1/api_test.rb

```diff
@@ -180,6 +179,96 @@ def app
     assert_empty json['data']
   end
 
+  # -------------------
+  # /search
+  # -------------------
+
+  test 'GET /v1/api/search fails if disabled key is specified' do
+    get "/api/v1/search?key=#{@disabled_key.key}&page=1"
+    assert_key_failed
+  end
+
+  test 'GET /v1/api/search returns 1st page with default params' do
+    get "/api/v1/search?key=#{@key.key}&page=1"
+    assert last_response.successful?
+    json = JSON.parse(last_response.body)
+    assert_equal 2, json['total_count']
+    data = json['data']
+    assert_equal 2, data.size
+    assert_manifestation(data[0], @manifestation, 'html', true)
+    assert_manifestation(data[1], @manifestation_2, 'html', true)
+  end
+
+  test 'GET /v1/api/search returns 1st page with descending popularity sorting, metadata view and epub format' do
+    get "/api/v1/search?key=#{@key.key}&page=1&sort_by=popularity&sort_dir=asc&view=metadata&file_format=epub"
+    assert last_response.successful?
+    json = JSON.parse(last_response.body)
+    assert_equal 2, json['total_count']
+    data = json['data']
+    assert_equal 2, data.size
+    assert_manifestation(data[0], @manifestation_2, 'epub', false)
+    assert_manifestation(data[1], @manifestation, 'epub', false)
+  end
+
+  test 'GET /v1/api/search applies filters' do
```

Comment:
where is the actual test here?

Reply:
Oops, fixed.

I started to writing this test and later switched to test for SearchManifestation service.

I've added couple of tests for API, one of the tests uses Mocha mock object simply to ensure that all supported parameters are correctly parsed and passed to SearchManifestation service correctly.

Also I've rebased branch on latest master.